### PR TITLE
Option to explicitly provide mardown file location for objects

### LINKED ISSUE #20405: MooseDocs doesn't find different markdown files for templated registered object
## Bug Description

When having multiple registered objects that are based on a templated version of class, there is no way to specify a different markdown file each object when doing the `!syntax list` command. It appears that MooseDocs just looks at the header file of the class and deduces the markdown file location. But in any situation where multiple objects are registered in a single file, they all reference the same markdown. 

## Steps to Reproduce

A good example is any AD capable object. Where the AD version and non-AD version point to the same markdown. This might be okay for these objects though, but in general.

## Impact

Object documentation is difficult to differentiate using `!syntax list` command and some documentation won't show up.

### PR DESCRIPTION
**Pull request by zachmprince**
Closes #20405

### CODE REVIEW

File: python/MooseDocs/extensions/appsyntax.py [hunk 1]
```diff
@@ -105,6 +105,8 @@ def defaultConfig():
         config['alias'] = (None, "Dictionary of syntax aliases.")
         config['unregister'] = (None,
                                 "A `dict` or `dict` of `dict` including syntax to unregister (key='moose_base', value='parent_syntax')")
+        config['markdown'] = (None,
+                             "A `dict` or `dict` of `dict` including markdown files to explicitly set markdown reference (key='syntax', value='file.md')")
```
Comment (cticenhour) [inline python/MooseDocs/extensions/appsyntax.py:109]:
```suggestion
        config['markdown'] = (None,
                             "A `dict` or `dict` of `dict` including markdown files to explicitly set a custom, additional markdown reference (key='syntax', value='file.md')")
```

Would like to make it clear in the description for developers that this is custom and in addition to the tree we already build.

**Review by cticenhour — CHANGES_REQUESTED** [review]
I like this well enough - especially since it allows app-level differences based on unique need. Would prefer in the long run something that can autodetect so we wouldn't need to manually set things, but this is not unlike the other customization we do in this extension. One suggested change noted.

**Comment by zachmprince** [comment]
Thanks @cticenhour. I committed your suggestion. I agree, auto-detection would be great. I had ideas for this including looking in multiple places based on the file name, class name, and moose-object name. But I couldn't resolve how to pick the right file if more than one existed. I figure this is probably the way to go for now, since it worked 99.9% of the time before this.

**Review by cticenhour — APPROVED** [review]

**Comment by cticenhour** [comment]
Sometimes, I think, it is best to start with a manual option just so we don't miss a corner case we didn't intend to. Thanks for tackling this @zachmprince!

**Pull request merged**